src/shared/components/data-table/components/table-controls/components/table-actions/components/download-data/DownloadData.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { ReactNode } from 'react';
+
+import useDataTable from 'src/shared/components/data-table/hooks/useDataTable';
+import { downloadTextAsFile } from 'src/shared/helpers/downloadAsFile';
+
+import { PrimaryButton } from 'src/shared/components/button/Button';
+
+import { TableAction } from 'src/shared/components/data-table/dataTableTypes';
+
+import styles from './DownloadData.scss';
+
+const getReactNodeText = (node: ReactNode): string => {
```

Comment:
There's a `renderToString` function exported from `'react-dom/server'`; I wonder if it won't be more suitable here.

(You can even pass the resulting string to a DOM element as `innerHTML` and request `textContent` back, I suppose; in order to strip all html tags that can be there).

Reply:
As a follow-up (continuing the discussion we had during scrum), I am still confused about whether this function will return the right thing for something like:

```
<div>
  <span>One</span>
  <span>Two</span>
</div>
```

From what I am seeing in the console (playing with someone else's code sandbox that uses React 16-something; but I doubt it will be different in React 18), `children` in such a case will be an array of react nodes, which will fail the first check (not string or number), fail the second check (no 'props' in array), and will return an empty string? I'm just running this in my head, so might be wrong.

When using the inbuilt browser methods, we won't need to think about all those edge cases.